server/migrations/20200903091227_data_migrate_default_branch_names.js

```diff
@@ -0,0 +1,16 @@
+
+exports.up = function(knex) {
+  	return knex('pushes')
+		.where('branch', 'master')
+		.update('branch', 'trunk')
+		.then(() => 
+			knex('ci_builds')
+				.where('branch', 'master')
+				.update('branch', 'trunk')
+		);
+};
+
+exports.down = function(knex) {
+	// It's unsafe to migrate this backwards
```

Comment:
Is it unsafe? Changing `trunk` back to `master` looks like an OK operation.

With this migration, I can imagine we might need to revert it.

Reply:
It would be unsafe in the future if we changed pushes that originated from trunk and never existed in master? Once the transition happens trunk will begin to move ahead of master.

Reply:
I'm not sure I understand 🙂 

Every database migration is unsafe in a sense that we can't freely revert it at any time and expect the app to continue working. If the migration adds a column, and the app starts to rely on it, then of course I can't revert the migration and delete the column.

When we talk about a migration being safe, I understand it as being reversible, not causing a loss of data. We can rename `master` to `trunk`, discover we made some serious mistake, need to revert, rename `trunk` back to `master` and the app can continue running. No data were lost.

An unsafe migration would be, e.g., dropping a table or column, or change data in a way that can't be reverted.